samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,412 @@
+
+	enum loc_accuracy gnss_accuracy = LOC_ACCURACY_LOW;
+	bool gnss_accuracy_set = false;
+
+	int cellular_timeout = 0;
+	bool cellular_timeout_set = false;
+
+	bool method1_set = false;
+	enum loc_method method1 = LOC_METHOD_CELLULAR;
+	bool method2_set = false;
+	enum loc_method method2 = LOC_METHOD_GNSS;
+
+	int opt;
+	int ret = 0;
+	int long_index = 0;
+	int method_count = 0;
+
+	if (argc < 2) {
+		goto show_usage;
+	}
+
+	/* command = argv[0] = "location" */
+	/* sub-command = argv[1]       */
+	if (strcmp(argv[1], "get") == 0) {
+		loc_cmd_args.command = LOCATION_CMD_GET;
+	} else if (strcmp(argv[1], "cancel") == 0) {
+		loc_cmd_args.command = LOCATION_CMD_CANCEL;
+	} else {
+		shell_error(shell, "Unsupported command=%s\n", argv[1]);
+		ret = -EINVAL;
+		goto show_usage;
+	}
+
+	/* Reset getopt due to possible previous failures */
+	optreset = 1;
+
+	/* We start from subcmd arguments */
+	optind = 2;
+
+	while ((opt = getopt_long(argc, argv, "", long_options, &long_index)) != -1) {
+		switch (opt) {
+		case LOCATION_SHELL_OPT_GNSS_TIMEOUT:
+			gnss_timeout = atoi(optarg);
+			if (gnss_timeout == 0) {
+				shell_error(shell, "GNSS timeout (%d) must be positive integer. ",
+					    gnss_timeout);
+				return -EINVAL;
+			}
+			gnss_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_CELLULAR_TIMEOUT:
+			cellular_timeout = atoi(optarg);
+			if (cellular_timeout == 0) {
+				shell_error(shell,
+					    "Cellular timeout (%d) must be positive integer. ",
+					    cellular_timeout);
+				return -EINVAL;
+			}
+			cellular_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_INTERVAL:
+			interval = atoi(optarg);
+			gnss_interval_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_GNSS_ACCURACY:
+			if (strcmp(optarg, "low") == 0) {
+				gnss_accuracy = LOC_ACCURACY_LOW;
+			} else if (strcmp(optarg, "normal") == 0) {
+				gnss_accuracy = LOC_ACCURACY_NORMAL;
+			} else if (strcmp(optarg, "high") == 0) {
+				gnss_accuracy = LOC_ACCURACY_HIGH;
+			} else {
+				shell_error(shell, "Unknown GNSS accuracy. See usage:");
+				goto show_usage;
+			}
+			gnss_accuracy_set = true;
+			break;
+		case LOCATION_SHELL_OPT_GNSS_NUM_FIXES:
+			gnss_num_fixes = atoi(optarg);
+			gnss_num_fixes_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_1:
+			if (strcmp(optarg, "cellular") == 0) {
+				method1 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method1 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method1_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_2:
+			if (strcmp(optarg, "cellular") == 0) {
+				method2 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method2 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method2_set = true;
+			break;
+		case '?':
```

Comment:
The syntax to get help is currently "location get --?". Not the most intuitive imo. Should we make it "location get help" or "location get ?"

Reply:
It should work like other commands, see for example: "link" / "link defcont". I.e. the help for different "levels".

Reply:
We've got few different styles in handling the help. The problem with this style in 'location' is that sometimes just the plain command does something and sometimes it gives you help.